src/Microsoft.DotNet.Interactive/KernelExtensions.cs

```diff
@@ -93,15 +93,29 @@ public static TKernel UseImportMagicCommand<TKernel>(this TKernel kernel)
         var command = new Command("#!import", "Imports and runs another notebook.");
         command.AddArgument(new Argument<FileInfo>("notebookFile").ExistingOnly());
         command.Handler = CommandHandler.Create(
-            async (FileInfo notebookFile, KernelInvocationContext _) =>
+            async (FileInfo notebookFile, KernelInvocationContext context) =>
             {
                 var document = await InteractiveDocument.LoadAsync(
                     notebookFile,
                     CreateKernelInfos(kernel.RootKernel as CompositeKernel));
 
                 foreach (var element in document.Elements)
                 {
-                    await kernel.RootKernel.SendAsync(new SubmitCode(element.Contents, element.KernelName));
+                    switch (element.KernelName?.ToLowerInvariant())
```

Comment:
At this point this should be based on the language name, not the kernel name.

Reply:
Correct, will do this in a new pr